internal/messagequeue/messagequeue.go

```diff
@@ -285,11 +283,42 @@ func (mq *MessageQueue) onShutdown() {
 func (mq *MessageQueue) runQueue() {
 	defer mq.onShutdown()
 
+	// Create a timer for debouncing scheduled work.
+	scheduleWork := time.NewTimer(0)
+	if !scheduleWork.Stop() {
+		<-scheduleWork.C
+	}
+
+	var workScheduled time.Time
 	for {
 		select {
 		case <-mq.rebroadcastTimer.C:
 			mq.rebroadcastWantlist()
-		case <-mq.outgoingWork:
+		case when := <-mq.outgoingWork:
```

Comment:
note: this solution _does_ have a drawback. If we have too much data to send and the next message isn't large enough to warrant an immediate send, we'll delay it.

Reply:
I'm not too concerned about that - it shouldn't happen often and in any case each want is independent. If `sendMessageMaxDelay` is short then it probably won't make much difference.